pkg/provider/core.go

```diff
@@ -101,13 +103,22 @@ func (p *Provider) CreateMachine(ctx context.Context, req *driver.CreateMachineR
 	}
 	instanceMetadata["user-data"] = string(userData)
 
+	folderId := string(secret.Data[api.YandexFolderID])
+
 	var networkInterfaceSpecs []*compute.NetworkInterfaceSpec
 	for _, netIf := range spec.NetworkInterfaceSpecs {
 		var natSpec *compute.OneToOneNatSpec
 		if netIf.AssignPublicIPAddress {
 			natSpec = &compute.OneToOneNatSpec{
 				IpVersion: compute.IpVersion_IPV4,
 			}
+
+			if len(netIf.PublicIPAddresses) > 0 {
+				natSpec.Address, err = FindUnusedAddressFromListByFolder(ctx, ySDK, folderId, netIf.PublicIPAddresses)
```

Comment:
The function call should be updated now :P
```suggestion
				natSpec.Address, err = findUnusedAddressFromListByFolder(ctx, ySDK, folderId, netIf.PublicIPAddresses)
```

Reply:
I'm so sorry, my bug. Fixed now. Thanks.